### PR TITLE
Multiple upgrades to get the system working locally with docker and Django 3

### PR DESCRIPTION
**Pull request by dukebody**
Update all needed packages to make the system work locally with modern Django 3.2.12 and docker.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -6,6 +6,6 @@ repos:
       - id: black
         language_version: python3
   - repo: https://github.com/pre-commit/mirrors-isort
-    rev: "master" # Use the revision sha / tag you want to point at
+    rev: "main" # Use the revision sha / tag you want to point at
```
Comment (dukebody) [inline .pre-commit-config.yaml:9]: project switched to `main` branch naming

File: Dockerfile [hunk 2]
```diff
@@ -1,27 +1,16 @@
-FROM python:3-alpine as builder
+FROM python:3.9-slim
```
Comment (dukebody) [inline Dockerfile:1]: Alpine doesn't always play well with Python. See https://pythonspeed.com/articles/alpine-docker-python/

File: Dockerfile [hunk 3]
```diff
@@ -1,27 +1,16 @@
...
 
 WORKDIR /home/python
 
-RUN apk add --no-cache zlib-dev build-base python-dev jpeg-dev
-RUN pip install virtualenv
-RUN virtualenv venv
```
Comment (dukebody) [inline Dockerfile:7]: No need to use venvs within Docker really.

File: README.md [hunk 4]
```diff
@@ -144,13 +144,13 @@ We use the following code formatters:
...
 
-On the first commit after installing `pre-commit`, Black and iSort it will create a new environment, which may take a few minutes. This environment will be reused for all subsequent commits.
+On the first commit after doing `pre-commit install`, Black and iSort it will create a new environment, which may take a few minutes. This environment will be reused for all subsequent commits.
 
 ### Loading City data
 
 In order to make it easier to create Places and Teams without having to manually
 enter records for Country, SPR (State/Province/Region) and City, you can preload
-them using data files from <http://download.geonames.org/export/dump/>
+them using data files from <https://download.geonames.org/export/dump/>
```
Comment (dukebody) [inline README.md:153]: Many browsers will prefer https over http.

File: docker-compose.yml [hunk 5]
```diff
@@ -4,11 +4,11 @@ services:
...
-    stdin_open: true
-    tty: true
     ports:
       - "8000:8000"
     environment:
       - DEBUG_MODE=True
       - SECRET_KEY=xxxxx
       - ALLOWED_HOSTS=localhost,127.0.0.1
+    volumes:
+      - ./:/home/python
```
Comment (dukebody) [inline docker-compose.yml:14]: Hot reload of changes without rebuilding the docker image.

File: events/templates/events/event_form.html [hunk 6]
```diff
@@ -1,3 +1,4 @@
 <table>
-{{ event_form }}
+    {{ event_form }}
+    {{ event_form.media }}
```
Comment (dukebody) [inline events/templates/events/event_form.html:3]: Needed for some widgets like recurrence.

File: get_together/settings.py [hunk 7]
```diff
@@ -232,6 +232,8 @@
     "images": {"nav_logo": "img/logo_b_v1.png"},
 }
 
+DEFAULT_AUTO_FIELD = "django.db.models.AutoField"
```
Comment (dukebody) [inline get_together/settings.py:235]: Needed in new Django to avoid warnings in `runserver`.

File: get_together/urls.py [hunk 8]
```diff
@@ -310,6 +317,7 @@
         views.show_team_events_by_slug,
         name="show-team-events-by-slug",
     ),
+    path('jsi18n.js', JavaScriptCatalog.as_view(packages=['recurrence']), name='jsi18n'),
```
Comment (dukebody) [inline get_together/urls.py:320]: Needed by `django-recurrence` to load `pgettext`.

File: requirements.txt [hunk 9]
```diff
@@ -3,18 +3,18 @@ certifi==2017.11.5
 chardet==3.0.4
 click==6.7
 decorator==4.2.1
-defusedxml==0.5.0
+defusedxml==0.6.0
 dj-database-url==0.4.2
-Django==2.0.12
-django-appconf==1.0.2
+Django==3.2.13
+django-appconf==1.0.5
```
Comment (dukebody) [inline requirements.txt:9]: Upgrades to work with Django 3 that doesn't include six.

File: resume/apps.py [hunk 10]
```diff
@@ -2,4 +2,4 @@
 
 
 class ResumeConfig(AppConfig):
-    name = "Resume"
+    name = "resume"
```
Comment (dukebody) [inline resume/apps.py:5]: Failed in Django 3 because it tried to import a module `Resume` that does not exist.

**Review by mhall119 — APPROVED** [review]
These all look good to me, thank you so much!

**Pull request merged**